recipes/xgboost/meta.yaml

```diff
@@ -0,0 +1,64 @@
+{% set name = "xgboost" %}
+{% set version = "0.6a2" %}
+{% set sha256 = "ebc4e2bf8c8266212e342ff8ec4f6ae469e8c41a05d099b6778de8424ce32563" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    - Makefile.patch  # [osx]
```

Comment:
OK. Well they already merged this. ☹️ 

We can try to convince them to just specify `clang` but I suspect more PRs on this might be annoying.

Here is the relevant snippet from their Makefile (w/ my PR merged):

```
# on Mac OS X, force brew gcc-6, since the Xcode c++ fails anyway
# it is useful for pip install compiling-on-the-fly
OS := $(shell uname)
ifeq ($(OS), Darwin)
export CC = $(if $(shell which gcc-6),gcc-6,$(if $(shell which gcc-mp-6), gcc-mp-6, $(if $(shell which gcc), gcc, clang)))
export CXX = $(if $(shell which g++-6),g++-6,$(if $(shell which g++-mp-6),g++-mp-6, $(if $(shell which g++), g++, clang++)))
endif
```

A few things:

1. They override any external settings of `CC` or `CXX` on macs. 
2. From the comments, it appears maybe they had trouble with with 'Xcode c++', whatever that is. This could all be OpenMP stuff.

 I think unless we convince them to just specify a single default per system and let people override it like normal in Makefiles, we will have to patch here.

However, I think we have gone off the rails completely. This discussion is best moved to an issue on xgboost about their build system on macs and letting people set their own compilers.

Since we have to patch anyways, we can just use what we want. It sounds like conda-forge prefers `clang` over `clang w/ gcc's c++ stdlib`, so I will modify the patch here to use `clang` only.